crates/apub/src/objects/instance.rs

```diff
@@ -178,19 +179,29 @@ impl Actor for ApubSite {
   }
 }
 
-/// try to fetch the instance actor (to make things like instance rules available)
-pub(in crate::objects) async fn fetch_instance_actor_for_object(
-  object_id: Url,
+/// Try to fetch the instance actor (to make things like instance rules available).
+pub(in crate::objects) async fn fetch_instance_actor_for_object<T: Into<Url> + Clone>(
+  object_id: &T,
   context: &LemmyContext,
   request_counter: &mut i32,
-) {
-  // try to fetch the instance actor (to make things like instance rules available)
+) -> Result<InstanceId, LemmyError> {
+  let object_id: Url = object_id.clone().into();
   let instance_id = Site::instance_actor_id_from_url(object_id);
   let site = ObjectId::<ApubSite>::new(instance_id.clone())
     .dereference(context, local_instance(context).await, request_counter)
     .await;
```

Comment:
This first tries to read the remote instance data from local db, and if its not found or its older than 24h, fetches it over http from the server. That's fine for Lemmy, but this will fail every time for most other software which doesn't have an instance actor, or has it at a different path. So there will be an unnecessary db read and http fetch every time, before we finally use the instance table directly below in the error case.

Not sure if there is a good way to optimize this situation. In any case db reads and http fetches are much less of a performance concern.

Reply:
Either that, or if other services have that actor at a specific url, add a nullable `instance_actor_url` column, that could be different from the domain. 

For the lemmy instance actors, we could copy all those over to the other column in a code migration, and make sure new inserts add to both.